components/widgets/menu/menu_wrapper.tsx

```diff
@@ -62,7 +63,26 @@ export default class MenuWrapper extends React.PureComponent<Props, State> {
         return null;
     }
 
+    public componentDidUpdate(prevProps: Props, prevState: State) {
+        if (this.state.open && !prevState.open) {
+            this.addEventListeners();
+        } else if (!this.state.open && prevState.open) {
+            this.removeEventListeners();
+        }
+    }
+
     public componentWillUnmount() {
+        if (this.state.open) {
+            this.removeEventListeners();
+        }
+    }
+
+    private addEventListeners() {
+        document.addEventListener('click', this.closeOnBlur, true);
```

Comment:
nit: out of the scope of the pr, but what about moving the `click` and `keyup` events names to a constant? (May be in this file)

Reply:
Yeah, we could consider that. I don't personally think it's that helpful though since they're standard values that come from the browser. I'm not opposed to doing that, but I'm too lazy to go back and do it here 😛